### PR TITLE
[azservicebus] Update readme, fix doc comments, etc...

### PR DESCRIPTION
**Pull request by richardpark-msft**
Some more doc comment and doc updates.

### CODE REVIEW

File: sdk/messaging/azservicebus/README.md [hunk 1]
```diff
@@ -113,48 +113,79 @@ Once you've created a [Client][godoc_client] you can create a [Sender][godoc_sen
...
-
-if err != nil {
-  panic(err)
-}
+Sending a single message:
 
-// send a single message
-err = sender.SendMessage(context.TODO(), &azservicebus.Message{
+```go
+// Send a single message
```
Comment (chlowell) [inline sdk/messaging/azservicebus/README.md:119]:
```suggestion
```
(redundant)

File: sdk/messaging/azservicebus/README.md [hunk 2]
```diff
@@ -113,48 +113,79 @@ Once you've created a [Client][godoc_client] you can create a [Sender][godoc_sen
...
 }, nil)
 ```
 
+#### Sending multiple messages using a batch
+
 You can also send messages in batches, which can be more efficient than sending them individually
 
 ```go
+// Sending multiple messages at a time using a MessageBatch
+
```
Comment (chlowell) [inline sdk/messaging/azservicebus/README.md:131]:
```suggestion
```

File: sdk/messaging/azservicebus/migrationguide.md [hunk 3]
```diff
@@ -43,6 +34,10 @@ New (using `azservicebus`):
 // new code
 
 client, err := azservicebus.NewClientFromConnectionString(connectionString, nil)
+
+//  or, if you would like to use an `azidentity` credential:
```
Comment (chlowell) [inline sdk/messaging/azservicebus/migrationguide.md:38]:
```suggestion
//  or, if you would like to use an azidentity credential:
```
But also consider splitting this into its own block so you can show more context, like importing `azidentity` and calling `NewDefaultAzureCredential`. That might help people getting started with track 2.

File: sdk/messaging/azservicebus/README.md [hunk 4]
```diff
@@ -113,48 +113,79 @@ Once you've created a [Client][godoc_client] you can create a [Sender][godoc_sen
...
+    // was full so it didn't go out with the previous SendMessageBatch call).
+    i--
+  } else if err != nil {
+    panic(err)
+  }
+}
+
+// check if any messages are remaining to be sent.
+if currentMessageBatch.NumMessages() > 0 {
+  err := sender.SendMessageBatch(context.TODO(), currentMessageBatch, nil)
```
Comment (chlowell) [inline sdk/messaging/azservicebus/README.md:184]: This is pretty big for a readme snippet. Could it be an example test instead?

**Review by chlowell — APPROVED** [review]

File: sdk/messaging/azservicebus/README.md (hunk 4 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/README.md:184]: It is a bit big, but my worry is that if someone copy/pastes they'll miss the pattern they really need to follow. I think we get a lot of copy/paste programming so I just wanted to be safe.

File: sdk/messaging/azservicebus/migrationguide.md (hunk 3 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/migrationguide.md]: Updated. I had a section later on in the doc specifically dedicated to the use of Azure Identity credentials, so I just linked to that.

**Pull request merged**